app.go

```diff
@@ -20,6 +23,8 @@ var (
 	errInvalidActionType = NewExitError("ERROR invalid Action type. "+
 		fmt.Sprintf("Must be `func(*Context`)` or `func(*Context) error).  %s", contactSysadmin)+
 		fmt.Sprintf("See %s", appActionDeprecationURL), 2)
+
+	showFlagError printerFunc = fmt.Fprintln
```

Comment:
I think this is hacky.
I am not familliar with the code base, but can you not wire up a memory writer in the test and check what was written?

Reply:
I agree that it's hacky 🙂

I can definitely achieve the same goal without this particular hacky part, I'll tear into that later today 🕐

Reply:
♻️ its been removed => https://github.com/urfave/cli/pull/819/commits/9438aba3b89e7053070ef277121a14e5fb95947e I'm working on upgrading the relevant tests now 📈